lib/floe/workflow/intrinsic_function.rb

```diff
@@ -2,13 +2,28 @@
 
 module Floe
   class Workflow
-    module IntrinsicFunction
-      def self.evaluate(payload, context = {}, input = {})
+    class IntrinsicFunction
+      def self.value(payload, context = {}, input = {})
+        new(payload).value(context, input)
+      end
+
+      def self.intrinsic_function?(payload)
+        payload.start_with?("States.")
+      end
+
+      attr_reader :payload
+
+      def initialize(payload)
+        @payload = payload
+      end
+
+      def value(context = {}, input = {})
         begin
           tree = Parser.new.parse(payload)
         rescue Parslet::ParseFailed => err
           raise Floe::InvalidWorkflowError, err.message
         end
```

Comment:
@Fryguy maybe in a follow-up but what do you think about doing the Parser.new.parse in the initialize?  It looks like it would always be the same for every invocation of value plus we like raising invalid workflow exception when building the Workflow instance not at runtime as much as possible